cli/tools/bench/mod.rs

```diff
@@ -415,49 +426,63 @@ pub async fn run_benchmarks(
   // Various bench files should not share the same permissions in terms of
   // `PermissionsContainer` - otherwise granting/revoking permissions in one
   // file would have impact on other files, which is undesirable.
-  let permission_desc_parser = factory.permission_desc_parser()?.clone();
+  let permission_desc_parser = factory.permission_desc_parser()?;
   let permissions = Permissions::from_options(
     permission_desc_parser.as_ref(),
     &cli_options.permissions_options(),
   )?;
+  let log_level = cli_options.log_level();
 
-  let members_with_bench_options =
-    cli_options.resolve_bench_options_for_members(&bench_flags)?;
-  let specifiers = members_with_bench_options
-    .iter()
-    .map(|(_, bench_options)| {
-      collect_specifiers(
-        bench_options.files.clone(),
-        cli_options.vendor_dir_path().map(ToOwned::to_owned),
-        is_supported_bench_path,
-      )
-    })
-    .collect::<Result<Vec<_>, _>>()?
-    .into_iter()
-    .flatten()
-    .collect::<Vec<_>>();
-
-  if specifiers.is_empty() {
-    return Err(generic_error("No bench modules found"));
+  let workspace_dirs_with_files = if cli_options.is_discovered_config() {
+    cli_options
+      .resolve_bench_options_for_members(&bench_flags)?
+      .into_iter()
+      .map(|(d, o)| (Arc::new(d), o.files))
+      .collect()
+  } else {
+    let patterns = bench_flags
+      .files
+      .as_file_patterns(cli_options.initial_cwd())?;
+    let config = cli_options.start_dir.to_bench_config(patterns)?;
+    let options = BenchOptions::resolve(config, &bench_flags);
+    vec![(cli_options.start_dir.clone(), options.files)]
+  };
```

Comment:
Why do we do different behaviour based on whether the config is discovered or not? Can we change `resolve_bench_options_for_members` to handle this instead?